### PR TITLE
MaryiaF_67322/fix: hide action buttons in the toolbox when the needed data …

### PR DESCRIPTION
**Pull request by maryiafrantsava-binary**
…(balance) is not loaded + refactoring component ToolBox

### CODE REVIEW

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx [hunk 1]
```diff
@@ -96,114 +118,119 @@ const ToolBox = ({ blockly }) => {
...
+        onClick={() => globalObserver.emit("blockly.start")}
+        classes={classNames("toolbox-button icon-run",{"toolbox-hide":!has_active_token || is_loading_balance})}
+        is_bot_running={is_bot_running}
       />
-      <Popover content={translate("Show log")} position="bottom">
-        <button id="logButton" className="toolbox-button icon-info" />
-      </Popover>
-      {has_active_token && <span className="toolbox-separator" />}
+      <ToolboxButton 
+        idContainer="stopButton"
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:206]:
```suggestion
        id_container="stopButton"
```

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx [hunk 2]
```diff
@@ -96,114 +118,119 @@ const ToolBox = ({ blockly }) => {
...
+        tooltip={translate("Stop the bot")}  
+        position="bottom"
+        onClick={()=>{globalObserver.emit("blockly.stop")}}
+        classes={classNames("toolbox-button icon-stop",{"toolbox-hide":!has_active_token})}
+      />
+      <ToolboxButton 
+        id={"logButton"}
+        classContainer={classNames({"toolbox-hide":!has_active_token})}
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:214]:
```suggestion
        class_container={classNames({"toolbox-hide":!has_active_token})}
```

File: src/botPage/view/deriv/components/popover/popover.jsx [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import React from 'react';
 import { Popover as TinyPopover } from 'react-tiny-popover';
 
-const Popover = ({ children , content })=>{
+const Popover = ({ id, classContainer, children , content })=>{
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/components/popover/popover.jsx:4]:
```suggestion
const Popover = ({ id, class_container, children , content })=>{
```

File: src/botPage/view/deriv/components/popover/popover.jsx [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
     const [is_hovered, updateIsHovered] = React.useState(false);
     
     return(
         <span
+            id={id}
+            className={classContainer}
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/components/popover/popover.jsx:10]:
```suggestion
            className={class_container}
```

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx [hunk 5]
```diff
@@ -23,19 +23,34 @@ const ShowModal = ({ modal, onClose, class_name }) => {
...
+  classes,
+  idContainer,
+  classContainer,
+  id,
+  onClick,
+  position = "bottom",
+  is_bot_running,
+}) => {
+  return (
+      <Popover id={idContainer} classContainer={classContainer} content={tooltip} position={position}>
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:38]:
```suggestion
      <Popover id={id_container} classContainer={class_container} content={tooltip} position={position}>
```

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx [hunk 6]
```diff
@@ -23,19 +23,34 @@ const ShowModal = ({ modal, onClose, class_name }) => {
...
-  <Popover content={tooltip} position={position}>
-    <button className={classes}>{label}</button>
-  </Popover>
-</span>
+const ToolboxButton = ({
+  label,
+  tooltip,
+  classes,
+  idContainer,
+  classContainer,
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:31]:
```suggestion
  id_container,
  class_container,
```

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 7]
```diff
@@ -49,7 +49,7 @@ const AnimateTrade = () => {
       setIndicatorMessage(INDICATOR_MESSAGES.starting);
       setContractStatus(CONTRACT_STATUS.not_running);
       globalObserver.emit("summary.disable_clear");
-      globalObserver.register("contract.status", animateStage);
+      globalObserver.register("contract.status", contract => animateStage(contract));
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:52]: There is no difference between 
`globalObserver.register("contract.status", contract => animateStage(contract))` 
and  
`globalObserver.register("contract.status", animateStage);`
you can revert this change

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx [hunk 8]
```diff
@@ -96,114 +118,119 @@ const ToolBox = ({ blockly }) => {
...
+        tooltip={translate("Run the bot")} 
+        position="bottom"
+        onClick={() => globalObserver.emit("blockly.start")}
+        classes={classNames("toolbox-button icon-run",{"toolbox-hide":!has_active_token || is_loading_balance})}
+        is_bot_running={is_bot_running}
       />
-      <Popover content={translate("Show log")} position="bottom">
-        <button id="logButton" className="toolbox-button icon-info" />
-      </Popover>
-      {has_active_token && <span className="toolbox-separator" />}
```
Comment (mehrzad-fs) [inline src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:193]: Why we don't use this method to show and hide the button and use class `toolbox-hide`?

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx (hunk 1 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:206]: done

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx (hunk 5 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:38]: done

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx (hunk 6 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:31]: done

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx (hunk 2 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:214]: done

File: src/botPage/view/deriv/components/popover/popover.jsx (hunk 3 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/components/popover/popover.jsx:4]: done

File: src/botPage/view/deriv/components/popover/popover.jsx (hunk 4 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/components/popover/popover.jsx:10]: done

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js (hunk 7 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/TradeInfoPanel/AnimateTrade.js:52]: ok

File: src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx (hunk 8 above)
Comment (maryiafrantsava-binary) [reply src/botPage/view/deriv/layout/ToolBox/ToolBox.jsx:193]: I reverted it. But `toolbox-hide` is saved in order to not display when important data hasn’t loaded yet.

**Review by mehrzad-fs — APPROVED** [review]

**Review by sara-fs — APPROVED** [review]
